### PR TITLE
Interface spec updates and more functionality

### LINKED ISSUE #43: Updates for v2-draft interface doc (Trial 1)
The changes for the new interface doc draft we received today makes some tweaks that we need to adjust our code for. Some of these are:
- Changes to Cueing messages.
- Changes to Telemetry messages and semantics.
- Better GEOjson compatibility.
- Field name changes, etc.
It added some changes which we already decided to do independently (replace TCP with HTTP, add a third Trial state, etc.).

### PR DESCRIPTION
**Pull request by ajfabbri**
Finishes up Start and End HTTP APIs for autonomy and base sides.

- Update spec documents with latest.
- Improve error handling to comply with spec.
- More unit testing.
- Coordinate geometry serde and unit tests (geoJSON)
- Various code cleanups.

Fixes #43 and #41

### CODE REVIEW

**Comment by ajfabbri** [comment]
Force push: fix lint issues

File: src/autov-cod/http_server.test.ts
Comment (kitplummer) [inline src/autov-cod/http_server.test.ts:1]: You have no idea how much appreciate not having to ask for tests, or justify the need.  Seriously.

**Review by kitplummer — COMMENTED** [review]
Tests fail for me locally - with this branch, out the box.  Just want to confirm they all pass for you locally @ajfabbri

File: src/autov-cod/http_server.test.ts
Comment (ajfabbri) [reply src/autov-cod/http_server.test.ts:1]: I feel seen. 🥹 Thank you!

**Comment by ajfabbri** [comment]
> Tests fail for me locally - with this branch, out the box. Just want to confirm they all pass for you locally @ajfabbri

Thanks for trying it out. Does work for me, checkout branch `npm run build` before `npm run test` (the test target doesn't build--wondering if maybe it should). Will ping you for error details.

**Review by kitplummer — APPROVED** [review]
LGTM.

NOTE: tests pass for me on macOS, fail on ubu-23.10.  Not a concern, just a note.

**Pull request merged**